builtin/logical/database/rotation_test.go

```diff
@@ -82,103 +82,168 @@ func TestBackend_StaticRole_Rotate_basic(t *testing.T) {
 		t.Fatalf("err:%s resp:%#v\n", err, resp)
 	}
 
-	data = map[string]interface{}{
-		"name":                "plugin-role-test",
-		"db_name":             "plugin-test",
-		"rotation_statements": testRoleStaticUpdate,
-		"username":            dbUser,
-		"rotation_period":     "5400s",
+	testCase := []struct {
+		name             string
+		rotationSchedule string
+		rotationPeriod   interface{}
+		rotationWindow   interface{}
```

Comment:
can we use `string` here or was there a reason these types needed to be `interface{}`?

Reply:
It look like these are interface{} so that we can set nil in the request data if they are unset in a given test case. Alternatively, we could make each test case define it's own data map?

Similar to: https://github.com/hashicorp/vault/blob/d6b7e5bfa1102acfd102b2000d3a6d944839eec7/builtin/logical/database/path_roles_test.go#L259-L262

Reply:
That's correct, I did it so we an set it to nil. If we prefer/like the approach of setting data maps for each test case better, I'd be happy to update it

Reply:
I think using a map might be a little more common for table tests and I probably prefer that style. But either way is ok

Reply:
Sounds good, I'll update it 👍